### ISSUE TITLE
#987 needs re-opening

### ISSUE THREAD

**Issue by robwatson-intechra** [angelleye/paypal-woocommerce#1006]
Please see https://github.com/angelleye/paypal-woocommerce/issues/987#issuecomment-366109735 for a potential reopening of #987

**Comment by iMansoorAliKhan** [comment]
Sorry for the delay in response, we are looking into that. Will get back to you soon.

**Comment by kcppdevelopers** [comment]
@iMansoorAliKhan I can't find any error on [this ](https://github.com/angelleye/paypal-woocommerce/files/1729852/payflow-orders-log.txt)log file.

**Comment by iMansoorAliKhan** [comment]
@kcppdevelopers , Did you see the original comment in other thread? user have showed some example, can you check in code if something from our code may be rounding off the amount  and skipping those decimal values?

**Comment by kcppdevelopers** [comment]
@iMansoorAliKhan yes, I have copy log file from in original thread, they show example but in log file it's seem it's working..

**Comment by robwatson-intechra** [comment]
Sorry, I forgot to attach this.

`2018-02-15T17:51:29+00:00 INFO PayFlow Request: Array
(
    [tender] => C
    [trxtype] => S
    [acct] => ****
    [expdate] => 0122
    [amt] => 92.00
    [currency] => USD
    [dutyamt] => 
    [freightamt] => 0.00
    [taxamt] => 0.00
    [taxexempt] => 
    [custom] => {"order_id":218931,"order_key":"wc_order_5a863920ac64b"}
    [comment1] => 
    [comment2] => 
    [cvv2] => ****
    [recurring] => 
    [swipe] => 
    [orderid] => 218931
    [orderdesc] => Order 218931
    [billtoemail] => xxxxxx@xxx.xxx
    [billtophonenum] => 
    [billtofirstname] => xxxxxxxxx
    [billtomiddlename] => 
    [billtolastname] => xxxxxxx
    [billtostreet] => xxxxx xxx xxxxxx 
    [billtocity] => xxxxxxxx
    [billtostate] => xx
    [billtozip] => xxxxx
    [billtocountry] => US
    [origid] => 
    [custref] => 
    [custcode] => 
    [custip] => xx.xxx.xx.xx
    [invnum] => xxxxxx
    [ponum] => 
    [starttime] => 
    [endtime] => 
    [securetoken] => 
    [partialauth] => 
    [authcode] => 
    [merchdescr] => xxxxxxxx xxxxxx
    [SHIPTOFIRSTNAME] => xxxxxxxxx
    [SHIPTOLASTNAME] => xxxxxxx
    [SHIPTOSTREET] => xxxxx xxx xxxxxx 
    [SHIPTOCITY] => xxxxxxxx
    [SHIPTOSTATE] => xx
    [SHIPTOCOUNTRY] => xx
    [SHIPTOZIP] => xxxxx
    [ITEMAMT] => 92.00
)

2018-02-15T17:51:32+00:00 INFO PayFlow Endpoint: https://payflowpro.paypal.com
2018-02-15T17:51:32+00:00 INFO PayFlow Response: Array
(
    [RESULT] => 0
    [PNREF] => BI1P774C87AA
    [RESPMSG] => Approved
    [AUTHCODE] => 111111
    [AVSADDR] => Y
    [AVSZIP] => Y
    [CVV2MATCH] => Y
    [PPREF] => 2G848242BG5471914
    [CORRELATIONID] => 9f4e0716dd87e
    [PROCAVS] => Y
    [PROCCVV2] => M
    [TRANSTIME] => 2018-02-15 17:51:29
    [BILLTOFIRSTNAME] => xxxxxxxxx
    [BILLTOLASTNAME] => xxxxxxx
    [AMT] => 92.00
    [ACCT] => xxxx
    [EXPDATE] => xxxx
    [CARDTYPE] => x
    [IAVS] => N
    [ORDERID] => 218931
    [RAWREQUEST] => BUTTONSOURCE[24]=AngellEYE_SP_WooCommerce&VERBOSITY[4]=HIGH&USER[16]=*****&VENDOR[14]=*****&PARTNER[6]=PayPal&PWD[15]=*****&TENDER[1]=C&TRXTYPE[1]=S&ACCT[16]=****&EXPDATE[4]=****&AMT[5]=92.00&CURRENCY[3]=USD&FREIGHTAMT[4]=0.00&TAXAMT[4]=0.00&CUSTOM[56]={"order_id":218931,"order_key":"wc_order_5a863920ac64b"}&CVV2[3]=****&ORDERID[6]=218931&ORDERDESC[80]=Order 218931&BILLTOEMAIL[14]=xxxxxx@xxx.xxx&BILLTOFIRSTNAME[9]=xxxxxxxxx&BILLTOLASTNAME[7]=xxxxxxx&BILLTOSTREET[17]=xxxxx xxx xxxxxx &BILLTOCITY[8]=xxxxxxxx&BILLTOSTATE[2]=xx&BILLTOZIP[5]=xxxxx&BILLTOCOUNTRY[2]=xx&CUSTIP[12]=xx.xxx.xx.xx&INVNUM[6]=218931&MERCHDESCR[15]=xxxxxxxx xxxxxx&SHIPTOFIRSTNAME[9]=xxxxxxxxx&SHIPTOLASTNAME[7]=xxxxxxx&SHIPTOSTREET[17]=xxxxx xxx xxxxxx &SHIPTOCITY[8]=xxxxxxxx&SHIPTOSTATE[2]=xx&SHIPTOCOUNTRY[2]=xx&SHIPTOZIP[5]=xxxxx&ITEMAMT[5]=92.00
    [RAWRESPONSE] => RESULT=0&PNREF=BI1P774C87AA&RESPMSG=Approved&AUTHCODE=111111&AVSADDR=Y&AVSZIP=Y&CVV2MATCH=Y&PPREF=2G848242BG5471914&CORRELATIONID=9f4e0716dd87e&PROCAVS=Y&PROCCVV2=M&TRANSTIME=2018-02-15 17:51:29&BILLTOFIRSTNAME=xxxxxxxxx&BILLTOLASTNAME=xxxxxxx&AMT=92.00&ACCT=xxxx&EXPDATE=xxxx&CARDTYPE=x&IAVS=N&ORDERID=218931
)`

**Comment by angelleye** [comment]
@robwatson-intechra The log you provided here seems to show a successful result..??

**Comment by robwatson-intechra** [comment]
Our fulfillment vendor is saying it came to them out of balance. They said what they got was:

Message: 
Order: 218931 
Error: order Total of 92.00 is not equal to the sum of line item, shipping, tax and discount applied charges of 92.0100 (Order.orderTotal)

**Comment by angelleye** [comment]
@robwatson-intechra We've been digging back into this again, and it's just not something that we're able to easily pinpoint what's going on here.  It seems you've got quite a bit of unique stuff in the mix, like 3 decimal pricing getting sent to 3rd party systems, and I'm struggling to follow exactly where the hiccup is happening.

Our plugin is processing the payment correctly, right?  The order shows up in WooCommerce as expected?  Is there anything wrong before data gets sent to your fulfillment center?

**Comment by robwatson-intechra** [comment]
It's been a while since I posted that one. I've not gotten notification of more chargebacks from the fulfillment center due to the decimal error. Let me check with them again to be sure. We may be okay now.

**Comment by robwatson-intechra** [comment]
The fulfillment center just told me that they are still getting the rounding errors in the orders but have been silently correcting them for us. But, they also have been charging a fee for each correction. 

I can't tell for sure whether the data is being altered before it leaves us or after it gets to the fulfillment center. I've looked through the Angell Eye code but just don't know enough to make that determination.

Assuming that this is not completely out of scope, I was wondering if I could grant access to your developer to our site so as to improve the debugging accuracy here. They would be able to review the logs without the obfuscations I've added to protect privacy when posting them here. They'd be able to see the whole ecosystem.

The only thing about the live environment is that it's read only for security purposes. If there's a need to change code to try different things, I can set up a dedicated development environment just for the developer to do whatever he wants.

**Comment by angelleye** [comment]
@robwatson-intechra What we need to do is get this moved into our new system.  Please submit a ticket here and reference ticket PFW-47 so we can work more directly on this with you.
